src/libraries/Common/src/System/Drawing/KnownColorTable.cs

```diff
@@ -466,11 +466,12 @@ internal static class KnownColorTable
         internal static Color ArgbToKnownColor(uint argb)
         {
             Debug.Assert((argb & Color.ARGBAlphaMask) == Color.ARGBAlphaMask);
-            Debug.Assert(s_colorValueTable.Length == ColorKindTable.Length);
+            Debug.Assert(ColorValueTable.Length == ColorKindTable.Length);
```

Comment:
(No action required on your part - just dropping this note as a general reminder.)

We'll want to confirm that `ColorValueTable.Length` is replaced with a literal int burned into the codegen. I remember the JIT needing to jump through some hoops to optimize the `((ROS<byte>)new byte[] { /* ... */ }).Length` case, and I suspect it was complex enough that we might not get it for free here.

Reply:
This:
```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

[DisassemblyDiagnoser(maxDepth: 3)]
public class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);

    private static ReadOnlySpan<char> Values => new char[] { 'a', 'b', 'c', 'd', 'e' };

    [Benchmark]
    public int GetLength() => Values.Length;
}
```
results in this IL for `Values`:
```
	IL_0000: ldtoken field valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=10' '<PrivateImplementationDetails>'::DC0F42A41F058686A364AF5B6BD49175C5B2CF3C4D5AE95417448BE3517B4008
	IL_0005: call valuetype [System.Runtime]System.ReadOnlySpan`1<!!0> [System.Runtime]System.Runtime.CompilerServices.RuntimeHelpers::CreateSpan<char>(valuetype [System.Runtime]System.RuntimeFieldHandle)
	IL_000a: ret
```
and this asm for GetLength:
```
## .NET 7.0.0 (42.42.42.42424), X64 RyuJIT
```assembly
; Program.GetLength()
       mov       eax,5
       ret
; Total bytes of code 6
```
So at least on this small repro, it looks good.

Reply:
That's awesome! Thanks for confirming.